src/orderbook/state.ts

```diff
@@ -101,23 +172,82 @@ export class AccountState {
             validUntil: parseInt(data.validUntil),
             priceNumerator: BigInt(data.priceNumerator),
             priceDenominator: BigInt(data.priceDenominator),
+            remainingAmount: BigInt(data.priceDenominator),
           })
           break
+        case 'SolutionSubmission':
+          /* eslint-disable no-case-declarations */
+          const submitter = data.submitter
+          const burntFees = BigInt(data.burntFees)
+          this.updateBalance(submitter, 0, amount => amount + burntFees)
+          lastSolution = { submitter, burntFees }
+          break
         case 'TokenListing':
           this.addToken(parseInt(data.id), data.token)
           break
-        case 'Deposit':
-        case 'SolutionSubmission':
         case 'Trade':
+          this.updateBalance(
+            data.owner,
+            parseInt(data.sellToken),
+            amount => amount - BigInt(data.executedSellAmount),
+          )
+          this.updateOrderRemainingAmount(
+            data.owner,
+            parseInt(data.orderId),
+            amount => amount - BigInt(data.executedSellAmount),
+          )
+          this.updateBalance(
+            data.owner,
+            parseInt(data.buyToken),
+            amount => amount + BigInt(data.executedBuyAmount),
+          )
+          break
         case 'TradeReversion':
+          if (lastSolution !== undefined) {
```

Comment:
We can't assert the batch IDs are the same because we only have the last batchId right?